src/entities/join/parser.ts

```diff
@@ -0,0 +1,24 @@
+import { BaseJoin } from '.';
+import { WeightedJoin } from './weighted/weightedJoin';
+
+/*********************** Basic Helper to get join class from pool type *************/
+export type JoinConfig = {
+    customPoolJoins: Record<string, BaseJoin>;
+};
+
+export class JoinParser {
+    private readonly poolJoins: Record<string, BaseJoin> = {};
+
+    constructor(config?: JoinConfig) {
+        const { customPoolJoins } = config || {};
```

Comment:
Also here, the idea behind supporting `customPoolJoins` is that I as the developer benefit somehow from using the SDK. That goal is not achieved with the current abstraction. I would still need to implement all functionality on my own.

If we take the SOR `customPoolFactories` as an example, by implementing a custom pool factory my pools are then included in the SOR routing.

Reply:
Looking at the weightedPool implementation, it feels like the only part that is poolType specific is the `userData` and everything else could be reused.
In any case, I'd say it will be easier to identify these opportunities once we implement a second join type. Then we rearrange methods to expect `customPoolJoins` to implement only what's necessary.
What do you think?